MoreLinq.Test/TrySingleTest.cs

```diff
@@ -0,0 +1,111 @@
+#region License and Terms
+// MoreLINQ - Extensions to LINQ to Objects
+// Copyright (c) 2019 James Webster. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#endregion
+
+namespace MoreLinq.Test
+{
+    using System;
+    using System.Collections.Generic;
+    using NUnit.Framework;
+
+    [TestFixture]
+    public class TrySingleTest
+    {
+        [Test]
+        public void TrySingleWithEmptySource()
+        {
+            var arrayWithNone = new int?[0];
+
+            var (cardinality, value) = arrayWithNone.TrySingle("zero", "one", "many", ValueTuple.Create);
```

Comment:
Let's use the simpler overload:

```suggestion
            var (cardinality, value) = arrayWithNone.TrySingle("zero", "one", "many");
```